msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByManagedIdentitySupplier.java

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class AcquireTokenByManagedIdentitySupplier extends AuthenticationResultSupplier {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AcquireTokenByManagedIdentitySupplier.class);
+
+    private ManagedIdentityParameters managedIdentityParameters;
+
+    AcquireTokenByManagedIdentitySupplier(ManagedIdentityApplication managedIdentityApplication, MsalRequest msalRequest) {
+        super(managedIdentityApplication, msalRequest);
+        this.managedIdentityParameters = (ManagedIdentityParameters) msalRequest.requestContext().apiParameters();
+    }
+
+    @Override
+    AuthenticationResult execute() throws Exception {
+
+        if (StringHelper.isNullOrBlank(managedIdentityParameters.resource)) {
+            throw new MsalClientException(
+                    MsalError.SCOPES_REQUIRED,
+                    MsalErrorMessage.SCOPES_REQUIRED);
+        }
+
+        TokenRequestExecutor tokenRequestExecutor = new TokenRequestExecutor(
+                clientApplication.authenticationAuthority,
+                msalRequest,
+                clientApplication.getServiceBundle()
+        );
+
+        if (!managedIdentityParameters.forceRefresh) {
```

Comment:
Did we expose this flag on MSAL.NET in the public API? I'd rather not make this public from the get go and see if ppl have good scenarios for it.

CC @localden

Reply:
We do expose force refresh in MSAL.Net for managed identity, [see](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForManagedIdentityParameterBuilder.cs#L56). Should we exclude it for MSAL Java?

Reply:
I think we should keep parity with MSAL .NET and keep the force refresh option.

Reply:
Agree with Den